www/js/controllers/home_controller.js

```diff
@@ -56,54 +59,79 @@
-          if(marker.lat !== null || marker.lng !== null){
-            vm.MapController.map.markers["Incident" + incident_num] = marker;
-          }
-          incident_num++;
-        });
-        vm.MapController.focusHere(LocationsService);
-      });;
-
+      vm.IncidentsController.getNewIncidents();
+      vm.MapController.focusHere(LocationsService);
     });
 
     /**
      * Center map on specific saved location
-     * @param locationKey
+     * @param incident
      */
-    vm.MapController.focusHere = function(incident) {
+    vm.MapController.focusHere = function (incident) {
       //if incident is starting point (Oahu), zoom out, if not, zoom in more to that location
       if (incident.starting === true) {
         var zoom_level = 11;
       } else {
         var zoom_level = 14;
       }
+      
+      //if the incident coordinates exist, center the map on that point
+      if(incident.lat !== null || incident.lng !== null){
+        vm.MapController.map.center  = {
+          lat : incident.lat,
+          lng : incident.lng,
+          zoom : zoom_level
+        };
+      }
 
-      //set map center
-      vm.MapController.map.center  = {
-        lat : incident.lat,
-        lng : incident.lng,
-        zoom : zoom_level
-      };
+      //If location is associated with a marker, open the popup
+      var marker = vm.MapController.findMarker(incident);
+      if (marker) {
+        marker.focus = true;
+      }
+
+      //FIX: Need error message for when there are no coordinates
     };
+    
+    /**
+     * Find marker associated with a specific location
+     * @param  {object} incident [incident object]
+     * @return {object} marker [marker associated with the location]
+     */
+    vm.MapController.findMarker = function (incident) {
+      var markers_obj = vm.MapController.map.markers;
+      for (var marker in markers_obj) {
+        if(markers_obj[marker].lat === incident.lat && markers_obj[marker].lng === incident.lng) {
+          return  markers_obj[marker];
+        }
+      }
+    }
+
+    vm.MapController.updateMarkers = function (incidents) {
+      vm.MapController.map.markers = {};
+      var incident_num = 1;
+      incidents.forEach(function (incident, index, array) {
+        //html for marker message
+        var html = "<h5 class='incident_type'>" + incident.type + "</h5>"
+          + "<p class='marker_text'>" + incident.date + "</p>"
+          + "<p class='marker_text'>" + incident.address + "</p>"
+          + "<p class='marker_text'>" + incident.area + "</p>";
+          // + "<button class='marker_text zoom' ng-click='focusHere(incident)'>Zoom Here</button>";
+        
+        //set marker location
+        var marker = {
+          lat: incident.lat,
+          lng: incident.lng,
+          message: html,
+        }
+
+        //Only add to list of markers if there are coordinates
+        if(marker.lat !== null || marker.lng !== null){
```

Comment:
Was thinking "||" because if either of the lat/lng is null the data is useless. If it's "&&" and only one of the them is null, that data point could sneak through